lib/core/CMakeLists.txt

```diff
@@ -1,19 +1,41 @@
-set(CORE_SOURCES
+add_library(AirshipCore)
+target_link_libraries(AirshipCore PUBLIC spdlog glfw AirshipCommonFlags)
+enable_clang_tidy(AirshipCore)
+
+set(AirshipCoreSources
     src/core/application.cpp
     src/core/event.cpp
 )
 
-add_library(core ${CORE_SOURCES})
-target_include_directories(core
+set(AirshipCoreHeaders
+    include/core/application.h
+    include/core/convar.h
+    include/core/event.h
+    include/core/logging.h
+    include/core/utils.hpp
+    include/core/window.h
+)
+
+target_include_directories(AirshipCore
     PUBLIC
         $<INSTALL_INTERFACE:include>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
-target_link_libraries(core PUBLIC spdlog glfw AirshipCommonFlags)
-target_compile_features(core PUBLIC cxx_std_20)
-target_compile_options(core PRIVATE
-  $<$<CXX_COMPILER_ID:MSVC>:/W4 /WX>
-  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic -Werror>
+
+target_sources(AirshipCore PRIVATE ${AirshipCoreSources})
+target_sources(AirshipCore PUBLIC FILE_SET HEADERS
+    BASE_DIRS include/core
```

Comment:
Is `BASE_DIRS` needed in other targets? e.g. `examples/shapes` doesn't have it.

Reply:
In some of the projects, header and cpp files are next to each other on disk, so they don't need BASE_DIRS definitions.